pkg/genericapiserver/filters/timeout.go

```diff
@@ -194,12 +196,8 @@ func (tw *baseTimeoutWriter) timeout(msg string) {
 	// handler
 	if !tw.wroteHeader && !tw.hijacked {
 		tw.w.WriteHeader(http.StatusGatewayTimeout)
-		if msg != "" {
-			tw.w.Write([]byte(msg))
-		} else {
-			enc := json.NewEncoder(tw.w)
-			enc.Encode(errors.NewServerTimeout(api.Resource(""), "", 0))
-		}
+		enc := json.NewEncoder(tw.w)
```

Comment:
Ugh, this isn't actually correct.  Timeout responses need to go through content type negotiation just like any other error.